qiskit/pulse/schedule.py

```diff
@@ -610,3 +643,44 @@ def bind_parameters(self, *args: List[Union[float, complex]],
     def __call__(self, *args: List[Union[float, complex]],
                  **kwargs: Dict[str, Union[float, complex]]) -> Schedule:
         return self.bind_parameters(*args, **kwargs)
+
+
+def _insertion_index(intervals: List[Interval], new_interval: Interval, index: int = 0) -> int:
```

Comment:
I believe an iterative binary search would be both a bit easier to reason about (no recursion)  and more performant. However, I think this is fine as is and will not block on it.